src/viking.js

```diff
@@ -1,11 +1,128 @@
 // Soldier
-function Soldier() {}
+function Soldier(health, strength) {
+    this.health = health
+    this.strength = strength
+
+    this.attack = function() {
+        return this.strength;
+    }
+    this.receiveDamage = function(theDamage) {
+        this.health -= theDamage;
+
+    }
+
+}
 
 // Viking
-function Viking() {}
+function Viking(name, health, strength) {
+
+    this.name = name;
+
+    Soldier.call(this, health, strength)
+
+    this.attack = function() {
```

Comment:
It's not necessary to redefine a method when it is equal to the one being inherited from the parent.